django/db/models/expressions.py

```diff
@@ -470,6 +470,27 @@ def desc(self):
         return OrderBy(self, descending=True)
 
 
+class ResolvedOuterRef(F):
+    """
+    An object that contains a reference to an outer query.
+
+    In this case, the reference to the outer query has been resolved, because the
+    inner query has been used as a subquery.
+    """
+    def as_sql(self, *args, **kwargs):
+        raise FieldError(
```

Comment:
Are these messages tested?
Also, I'm not sure they're helpful enough that users will be able to understand how to correct the issue.